internal/schemautil/userconfig/apiconvert/toapi.go

```diff
@@ -0,0 +1,281 @@
+package apiconvert
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil/userconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var typedKeyRegexp = regexp.MustCompile(`^[A-z]*_(boolean|integer|number|string|array|object)$`)
```

Comment:
Just noticing this is not equal to `a-zA-Z_-`. It's shorter but has more cases also.

Reply:
changed it to match `.*`